src/linter/cache_test.go

```diff
@@ -86,6 +86,15 @@ function as_shape($x) {
   return $x;
 }
 
+class Foo {}
+
+function infer_types_in_callback() {
+  /** @var Foo[] $a */
+  $a = [];
+
+  array_map(function($item) {}, $a);
```

Comment:
I would like the array_map result to be used. Tests should have valid code that doesn't trigger any warnings (preferably).